bundler/lib/bundler/shared_helpers.rb

```diff
@@ -48,15 +48,11 @@ def in_bundle?
     end
 
     def chdir(dir, &blk)
-      Bundler.rubygems.ext_lock.synchronize do
-        Dir.chdir dir, &blk
-      end
+      Dir.chdir dir, &blk
```

Comment:
i think this is a change in behavior -- it might lead to conflicting `chdir`s when loading gemspecs. i think it still needs to be guarded by a mutex (same thing with `pwd` below)

Reply:
You're right. The mutex name was a bit unfortunately because it suggested that we only chdir when building extensions, which we no longer do. But we indeed still chdir when loading gemspecs, and this removes syncronizing that. I'll restore the monitor.

Reply:
I restored the monitor, but left any naming improvements for another time.